python/pyiceberg/expressions/__init__.py

```diff
@@ -32,6 +41,24 @@
 B = TypeVar("B")
 
 
+def _to_literal(lit: Optional[Union[T, Literal[T]]]) -> Optional[Literal[T]]:
+    # This should not be optional, to be fixed in a separate PR
+    if lit:
+        return lit if isinstance(lit, Literal) else literal(lit)
+    else:
+        return None
+
+
+def _combine_into_set(values: Tuple[Union[T, Literal[T], Iterable[T], Iterable[Literal[T]]], ...]) -> Set[Literal[T]]:
```

Comment:
Do we need values to be so complex? I would say that we can accept any `Iterable[T]` or `Iterable[Literal[T]]`. We don't need to support mixing literals with raw values, right?

Reply:
From the signature of the `__new__` function of `{BoundSetPredicate,SetPredicate}` you can see that you can also provide positional arguments, I think that's a nice touch:
```python
def test_in_positional_args():
    assert In(Reference("foo"), "a", "bc", "def").literals == {StringLiteral("a"), StringLiteral("bc"), StringLiteral("def")}
```

Reply:
I'm fine with positional args (or not) but what I don't think we need is support for mixing literals with values:

```python
    In("str_col", "a", literal("b")) # seems unnecessarily complicated
    In("str_col", "a", "b") # fine
    In("str_col", literal("a"), literal("b")) # also fine
```